clients/banking/src/components/BeneficiaryInternationalWizardForm.tsx

```diff
@@ -81,19 +97,49 @@ export const TransferInternationalWizardBeneficiary = ({
     name: { initialValue: initialBeneficiary?.name ?? "", validate: validateRequired },
   });
 
+  const currencyItems = useMemo(() => {
+    return currencies.map(value => {
+      const name = currencyResolver?.of(value);
+      const code = currencyFlags[value];
+
+      return {
+        icon: <Flag code={code} width={18} />,
+        name: isNotNullish(name) ? `${value} (${name})` : value,
+        value,
+      };
+    });
+  }, []);
+
   const handleOnResultsChange = useDebounce<ResultItem[]>(value => {
     const nextResults = value.filter(({ value }) => isNotEmpty(value));
     setResults(nextResults);
     setVariables({ dynamicFields: nextResults });
   }, 1000);
 
+  const handleOnCurrencyChange = useCallback(
+    (currency: Currency) => {
+      setCurrency(currency);
+      setRoute(Option.None()); // reset route state
+
+      setVariables({
+        currency,
+        amountValue: match(currency)
+          .with("IDR", "VND", () => "50000")
+          .otherwise(() => "1000"),
+      });
```

Comment:
rather than implementing a `getVariables`, how about turning this into the following?

```ts
useEffect(() => {
  setVariables({
    currency,
    amountValue: match(currency)
      .with("IDR", "VND", () => "50000")
      .otherwise(() => "1000"),
  });
}, [currency])
```

Reply:
This doesn't solve the issue:
- We still have to declare an extra `currency` state that **cannot** be used in query variables (or it will trigger loading state again)
- This is still triggering two renders
- Calling `setVariables` in a `useEffect` instead of the callback just delay the network call by one render